h2o-persist-hdfs/src/main/java/water/persist/PersistHdfs.java

```diff
@@ -79,6 +75,7 @@
         Log.debug("Cannot find HADOOP_CONF_DIR or YARN_CONF_DIR - default HDFS properties are NOT loaded!");
       }
     }
+    conf.set("fs.s3a.aws.credentials.provider", "water.persist.H2ODynamicS3ACredentialsProvider,com.amazonaws.auth.EnvironmentVariableCredentialsProvider,com.amazonaws.auth.InstanceProfileCredentialsProvider,org.apache.hadoop.fs.s3a.AnonymousAWSCredentialsProvider");
```

Comment:
But what if the users have their own configuration? This would overwrite it

Reply:
What is meant by "their own configuration" ? 

There is:
- H2ODynamicS3ACredentialsProvider (ours)
- EnvironmentVariableCredentialsProvider
- InstanceProfileCredentialsProvider
- AnonymousAWSCredentialsProvider

Is there any other missing ?

Reply:
The user can specify anything (including their custom provider) in Hadoop configuration for the property "fs.s3a.aws.credentials.provider". Different Hadoop distributions can have different default providers, I could imagine eg. EMR will have very different providers.

My point was that we need to respect what the user has specified in their Hadoop configuration and avoid overwriting something unintentionally.

Reply:
For other reviewers: I'll try to find out if simply detecting this value is set to anything at runtime and then avoiding overriding it with our provider chain is enough, or there is any caveeat to it.

Reply:
@michalkurka 

Once the configuration file (in this simple case the default one `etc/hadoop/core-site.xml`) is modified to look like this:

```xml
<configuration>
	<property>
		<name>fs.s3a.aws.credentials.provider</name>
		<value>
			org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider,
			com.amazonaws.auth.EnvironmentVariableCredentialsProvider,
			com.amazonaws.auth.InstanceProfileCredentialsProvider
		</value>
	</property>
</configuration>
```

The value is detectable at the time the CONF object is constructed. Which means we can detect changes to this value and not modify it if user has previously set it to a different value in the configuration file. Let me demonstrate it simply with a screenshot:

![Snímek z 2019-05-19 13-11-39](https://user-images.githubusercontent.com/8769110/57981441-df521800-7a37-11e9-93e3-4730b518d2af.png)